src/main/java/uk/gov/hmcts/ccd/domain/model/definition/CaseDetails.java

```diff
@@ -356,4 +355,21 @@ public boolean hasCaseReference() {
         return getReference() != null;
     }
 
+    @JsonIgnore
+    public Map<String, JsonNode> getCaseEventData(CaseEventDefinition caseEventDefinition) {
```

Comment:
if you notice the previous version of CaseDetails it knows nothing about event, event definition, event field definitions, it is not tied to event classes like CaseEventDefinition. I think it would be better that way and make this method more generic, reusable and easily testable in isolation, e.g.

`public Map<String, JsonNode> getFields(String... ids) {}`

This way we can use to get field values even when the fields do not come from an event...

Reply:
Makes sense.. Will change this

Reply:
changed this to string array